pandas/core/arrays/arrow/array.py

```diff
@@ -258,9 +258,19 @@ def _from_sequence_of_strings(
 
             scalars = to_datetime(strings, errors="raise").date
         elif pa.types.is_duration(pa_type):
-            from pandas.core.tools.timedeltas import to_timedelta
+            try:
+                # GH51175: test_from_sequence_of_strings_pa_array
+                # attempt to parse as int64 reflecting pyarrow's
+                # duration to string casting behavior
+                if isinstance(strings, (pa.Array, pa.ChunkedArray)):
+                    scalars = strings
+                else:
+                    scalars = pa.array(strings, type=pa.string(), from_pandas=True)
```

Comment:
IMO (slightly) I think we should still prefer to use `to_timedelta` first if a non-pyarrow `scalars` was passed over the `cast`

Reply:
I think using `to_timedelta` first would be incorrect in some cases. For instance, if passed "1000 seconds" as
```
ArrowExtensionArray._from_sequence_of_strings(["1000"], dtype=pa.duration("s"))
```

`pd.to_timedelta(["1000"])` is interpreted as `1000 ns` (`to_timedelta` does not allow specifying `unit` with a string input and assumes nanoseconds).

Reply:
in fact, the example provided above errors on main due to assuming ns units:

```
import pyarrow as pa

ArrowExtensionArray._from_sequence_of_strings(["1000"], dtype=pa.duration("s"))

# ArrowInvalid: Casting from duration[ns] to duration[s] would lose data: 1000
```

Reply:
FWIW there's a corner case with iNaT that pyarrow will handle differently from to_timedelta

Reply:
Good point re: `NaT`. I made some changes which now call `to_timedelta` first and only attempts the int conversion when the pyarrow unit is non-nano. It respects the null mask from `to_timedelta` so I think the `NaT` edge case is handled. I added a test for it. 

It all feels a bit messy, but I don't see another option ATM. Open to suggestions.